src/plccng/scanner/scanner_test.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+
+from ..lines import parseLines
+from .scanner import Scanner
+from .structs import LexError, Line, Skip, Token
+
+
+@pytest.fixture
+def errorRaisingMatcher():
+    class ErrorRaisingMatcher:
+        def match(self, line, index):
+            return LexError(line=line, column=index)
+
+    return ErrorRaisingMatcher()
+
+@pytest.fixture
+def whiteSpaceSkipMatcher():
+    class WhiteSpaceSkipMatcher():
+        def match(self, line, index):
+            if line.string[index] == " ":
+                return Skip(name="whitespace", lexeme=" ", line=line, column=index)
+            else:
+                return LexError(line=line, column=index)
+    
+    return WhiteSpaceSkipMatcher()
+
+#Matches up to 5 of any characters (whitespace included) in a string
+@pytest.fixture
+def fiveCharacterMatcher():
+    class FiveCharacterMatcher():
+        def match(self, line, index):
+            final_index = len(line.string)
+            if index+5 > final_index:
+                return Token(name="Token", lexeme=line.string[index:final_index], line=line, column=final_index)
+            return Token(name="Token", lexeme=line.string[index:index+5], line=line, column=index)
+    
+    return FiveCharacterMatcher()
+
+    
+
+
+def test_no_lines_given_returns_nothing():
+    scanner = Scanner(matcher=None)
+    results = scanner.scan(None)
+    assert list(results) == []
+
+
+def test_empty_lines_given_returns_nothing():
+    scanner = Scanner(matcher=None)
+    results = scanner.scan([])
+    assert list(results) == []
+
+
+def test_LexError(errorRaisingMatcher):
+    scanner = Scanner(errorRaisingMatcher)
+    oneLineWithError = parseLines('''\
+blah blah
+''')
+    result = list(scanner.scan(oneLineWithError))
+    assert isinstance(result[0], LexError)
+
+
+def test_LexError_at_start_moves_on_to_next_line(errorRaisingMatcher):
+    scanner = Scanner(errorRaisingMatcher)
+    twoLinesWithErrors = parseLines('''\
+blah blah
+next line
+''')
+    scanner = Scanner(errorRaisingMatcher)
+    results = list(scanner.scan(twoLinesWithErrors))
+    assert isinstance(results[0], LexError)
+    assert results[1].line.string == 'next line'
+
+def test_lex_error_in_middle_of_the_line_moves_on_to_the_next_line(whiteSpaceSkipMatcher):
+    lineWithErrorinTheMiddle = parseLines('''\
+ this line should create a skip first
+after the skip and lexerror this line should also be a lex error
+''')
+    scanner = Scanner(whiteSpaceSkipMatcher)
+    results = list(scanner.scan(lineWithErrorinTheMiddle))
+    assert isinstance(results[0], Skip)
+    assert isinstance(results[1], LexError)
+    assert results[2].line.string == 'after the skip and lexerror this line should also be a lex error'
+
+def test_indexes_are_correct_after_matches(fiveCharacterMatcher):
```

Comment:
What indexes must be correct?

In the test below, the only index I see is in `result.lexeme[0] == '1'`.
But I don't think it's '0' that should be correct. In fact, this line should be rewritten as `result.lexeme == '12345'`.

So what are we testing here?